TM1py/Services/ApplicationService.py

```diff
@@ -15,143 +16,161 @@ def __init__(self, tm1_rest):
+
+        if not application_type == ApplicationTypes.FOLDER:
+            name += application_type.suffix
+
+        contents = 'PrivateContents' if private else 'Contents'
+        mid = "".join(["/Contents('{}')".format(element) for element in path.split('/')])
+        base_url = "/api/v1/Contents('Applications'){dynamic_mid}/{contents}('{application_name}')".format(
+            dynamic_mid=mid,
+            contents=contents,
+            application_name=name)
+
+        if application_type == ApplicationTypes.CUBE:
+            response = self._rest.GET(base_url + "?$expand=Cube($select=Name)")
+            return CubeApplication(path=path, name=name, cube_name=response.json()["Cube"]["Name"])
+
+        elif application_type == ApplicationTypes.CHORE:
+            response = self._rest.GET(base_url + "?$expand=Chore($select=Name)")
+            return ChoreApplication(path=path, name=name, chore_name=response.json()["Chore"]["Name"])
+
+        elif application_type == ApplicationTypes.DIMENSION:
+            response = self._rest.GET(base_url + "?$expand=Dimension($select=Name)")
+            return DimensionApplication(path=path, name=name, dimension_name=response.json()["Dimension"]["Name"])
+
+        elif application_type == ApplicationTypes.FOLDER:
+            # implicit TM1pyException if doesn't exist
+            self._rest.GET(base_url)
+            return FolderApplication(path=path, name=name)
+
+        elif application_type == ApplicationTypes.LINK:
+            # implicit TM1pyException if doesn't exist
+            self._rest.GET(base_url)
+            response = self._rest.GET(base_url + "?$expand=*")
+            return LinkApplication(path=path, name=name, url=response.json()["URL"])
+
+        elif application_type == ApplicationTypes.PROCESS:
+            response = self._rest.GET(base_url + "?$expand=Process($select=Name)")
+            return ProcessApplication(path=path, name=name, process_name=response.json()["Process"]["Name"])
+
+        elif application_type == ApplicationTypes.SUBSET:
+            response = self._rest.GET(
+                base_url +
+                "?$expand=Subset($select=Name;$expand=Hierarchy($select=Name;$expand=Dimension($select=Name)))")
+            return SubsetApplication(
+                path=path,
+                name=name,
+                dimension_name=response.json()["Subset"]["Name"],
+                hierarchy_name=response.json()["Subset"]["Name"],
+                subset_name=response.json()["Subset"]["Name"])
+
+        elif application_type == ApplicationTypes.VIEW:
+            response = self._rest.GET(base_url + "?$expand=View($select=Name;$expand=Cube($select=Name))")
+            return ViewApplication(
+                path=path,
+                name=name,
+                cube_name=response.json()["View"]["Cube"]["Name"],
+                view_name=response.json()["View"]["Name"])
+
+    def get_document(self, path, name, private=False):
```

Comment:
Let's make it even more consistent with the rest of the TM1py code.
All 'old' methods, for example, the get or update in SubsetService assuming the private parameter as optional and True as default. I suggest to synchronize it and make it consistent. Either everywhere 'private=False' (my favorite :) ) or the other way around.

Reply:
I like it.
Will change the other `private` default values to `False`